src/test/java/com/mongodb/InterruptibilityItTest.java

```diff
@@ -0,0 +1,277 @@
+                        .connectTimeout(toIntExact(blockTimeout.toMillis()), MILLISECONDS)
+                        .readTimeout(toIntExact(blockTimeout.toMillis()), MILLISECONDS)
+                )
+                .applyToClusterSettings(builder -> builder
+                        .mode(ClusterConnectionMode.MULTIPLE)
+                        .requiredClusterType(ClusterType.REPLICA_SET)
+                        .serverSelectionTimeout(blockTimeout.toMillis(), MILLISECONDS)
+                );
+        clientSettings = clientSettingsBuilderSupplier.get().build();
+        client = MongoClients.create(clientSettings);
+        collection = client.getDatabase(InterruptibilityItTest.class.getSimpleName()).getCollection("collection");
+    }
+
+    @AfterAll
+    static void afterAll() {
+        client.close();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        InterruptibilityItTest.collection.drop();
+    }
+
+    @AfterEach
+    void afterEach() {
+        // make sure the current thread is not interrupted because tests mess with this status
+        com.mongodb.assertions.Assertions.assertFalse(Thread.interrupted());
+        // make sure the primary is available, because tests mess with it
+        primary(client, Duration.ofNanos(Long.MAX_VALUE));
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void mongoCollectionDrop(final Duration interruptDelay) {
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, "drop", collection::drop);
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void mongoIterableFirst(final Duration interruptDelay) {
+        MongoIterable<String> iterable = client.listDatabaseNames();
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, "listDatabases", iterable::first);
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void mongoCursorNext(final Duration interruptDelay) {
+        collection.insertMany(IntStream.range(0, 2).mapToObj(i -> new Document()).toList());
+        try (MongoCursor<Document> cursor = collection.find().batchSize(1).cursor()) {
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 1);
+            cursor.next();
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 0);
+            assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                    interruptDelay, "getMore", cursor::next);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void serverSelection(final Duration interruptDelay) {
+        // we assume here that the time to elect a new leader is greater than `interruptDelay`
+        client.getDatabase("admin").runCommand(BsonDocument.parse(
+                "{\n"
+                // using `blockTimeout` does not give us much because it controls only the unelectability of the current primary
```

Comment:
Not sure what this comment is intended to convey.  Does `blockTimeout` give us what we need? If so, it seems unnecessary to say that it doesn't give us much.  Please clarify, or remove.

Reply:
Done.